backend/parser/listener/listener.go

```diff
@@ -0,0 +1,211 @@
+		bodyColumn:       bodyColumn,
+		attributesColumn: attributesColumn,
+		keysToColumns:    keysToColumns,
+	}
+}
+
+func (s *searchListener) EnterSearch_query(ctx *parser.Search_queryContext) {}
+func (s *searchListener) ExitSearch_query(ctx *parser.Search_queryContext) {
+	s.sb.Where(s.rules...)
+}
+
+func (s *searchListener) EnterTop_paren_col_expr(ctx *parser.Top_paren_col_exprContext) {}
+func (s *searchListener) ExitTop_paren_col_expr(ctx *parser.Top_paren_col_exprContext)  {}
+
+func (s *searchListener) EnterNegated_top_col_expr(ctx *parser.Negated_top_col_exprContext) {}
+func (s *searchListener) ExitNegated_top_col_expr(ctx *parser.Negated_top_col_exprContext)  {}
+
+func (s *searchListener) EnterTop_col_search_value(ctx *parser.Top_col_search_valueContext) {}
+func (s *searchListener) ExitTop_col_search_value(ctx *parser.Top_col_search_valueContext)  {}
+
+func (s *searchListener) EnterCol_paren_expr(ctx *parser.Col_paren_exprContext) {}
+func (s *searchListener) ExitCol_paren_expr(ctx *parser.Col_paren_exprContext)  {}
+
+func (s *searchListener) EnterNegated_col_expr(ctx *parser.Negated_col_exprContext) {}
+func (s *searchListener) ExitNegated_col_expr(ctx *parser.Negated_col_exprContext) {
+	rule := s.rules[len(s.rules)-1]
+	s.rules = s.rules[:len(s.rules)-1]
+	s.rules = append(s.rules, fmt.Sprintf("NOT (%s)", rule))
+}
+
+func (s *searchListener) EnterAnd_col_expr(ctx *parser.And_col_exprContext) {}
+func (s *searchListener) ExitAnd_col_expr(ctx *parser.And_col_exprContext) {
+	rules := s.rules[len(s.rules)-2:]
+	s.rules = s.rules[:len(s.rules)-2]
+	s.rules = append(s.rules, s.sb.And(rules...))
+}
+
+func (s *searchListener) EnterOr_col_expr(ctx *parser.Or_col_exprContext) {}
+func (s *searchListener) ExitOr_col_expr(ctx *parser.Or_col_exprContext) {
+	rules := s.rules[len(s.rules)-2:]
+	s.rules = s.rules[:len(s.rules)-2]
+	s.rules = append(s.rules, s.sb.Or(rules...))
+}
+
+func (s *searchListener) EnterCol_search_value(ctx *parser.Col_search_valueContext) {}
+func (s *searchListener) ExitCol_search_value(ctx *parser.Col_search_valueContext)  {}
+
+func (s *searchListener) EnterNegated_search_expr(ctx *parser.Negated_search_exprContext) {}
+func (s *searchListener) ExitNegated_search_expr(ctx *parser.Negated_search_exprContext) {
+	rule := s.rules[len(s.rules)-1]
+	s.rules = s.rules[:len(s.rules)-1]
+	s.rules = append(s.rules, fmt.Sprintf("NOT (%s)", rule))
+}
+
+func (s *searchListener) EnterBody_search_expr(ctx *parser.Body_search_exprContext) {
+	s.currentKey = s.bodyColumn
+	s.currentOp = "="
+}
+func (s *searchListener) ExitBody_search_expr(ctx *parser.Body_search_exprContext) {}
+
+// There is an issue here when grouping clauses together that contain a LIKE.
+// e.g. a:b* a!=bc a!=bb will result in a query like:
+// SELECT * FROM t WHERE ((a LIKE 'b%' AND a <> 'bc') AND a <> 'bb')
+// Commented out for now, but we will probably want to support something like
+// this in the future.
+// TODO: Review w/ Zane.
```

Comment:
@mayberryzane wanted to look at this one with you. Seems like we want this grouping from `ExitAnd_search_expr`, but only when we encounter an explicit `AND` clause - would you agree?

Reply:
I ended up implementing this in e2add65322c380ebfef321bc089a34d319f2bf3e - let me know if you think that's not the right approach!